src/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -187,20 +183,9 @@ public class DataController : ControllerBase
                     bool parseSuccess = Request.Headers.TryGetValue("Content-Disposition", out StringValues headerValues);
                     string? filename = parseSuccess ? DataRestrictionValidation.GetFileNameFromHeader(headerValues) : null;
 
-                    IEnumerable<FileAnalysisResult> fileAnalysisResults = new List<FileAnalysisResult>();
-                    if (FileAnalysisEnabledForDataType(dataTypeFromMetadata))
-                    {
-                        fileAnalysisResults = await _fileAnalyserService.Analyse(dataTypeFromMetadata, fileStream, filename);
-                    }
-
-                    bool fileValidationSuccess = true;
-                    List<ValidationIssue> validationIssues = new();
-                    if (FileValidationEnabledForDataType(dataTypeFromMetadata))
-                    {
-                        (fileValidationSuccess, validationIssues) = await _fileValidationService.Validate(dataTypeFromMetadata, fileAnalysisResults);
-                    }
+                    var validationIssues = await _validationService.ValidateFileUpload(instance, dataTypeFromMetadata, fileStream.ToArray(), filename, streamContent.Headers.ContentType!.ToString(), language);
```

Comment:
Intead of non-null assertion on `streamContent.Headers.ContentType`, should we return a bad request if there is no content type header? Unless that is already being done somewhere?

Reply:
Nvm, seems like `mimeType` is nullable on `ValidateFileUpload`, so I think it should be `?` instead?

Reply:
Probably.
```suggestion
                    var validationIssues = await _validationService.ValidateFileUpload(instance, dataTypeFromMetadata, fileStream.ToArray(), filename, streamContent.Headers.ContentType?.ToString(), language);
```
I was under the impression that `DataRestrictionValidation.CompliesWithDataRestrictions` on line 160 would ensure that this was available, but it looks like I was wrong.